### PR TITLE
Added telemetry classes to produce inline publish functionality

### PR DESCRIPTION
**Pull request by ismayilov-ismayil**
**Prob:** We need to have the possibility to publish telemetry from Agent.Listener code as well.
Now this method is available only from the Agent.Worker.
**Solution:**. Added corresponding classes to provide telemetry data publishing availability in Agent.Listener

### CODE REVIEW

File: src/Agent.Listener/Telemetry/TelemetryPublisher.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    public sealed class TelemetryPublisher : AgentService, IAgenetListenerTelemetryPublisher
+    {
+        private ICustomerIntelligenceServer _ciService;
+
+        public string Name => "publish";
+        public List<string> Aliases => null;
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Maintainability", "CA2000:Dispose objects before losing scope", MessageId = "jobServer")]
```
Comment (KonstantinTyukalov) [inline src/Agent.Listener/Telemetry/TelemetryPublisher.cs:31]: Can we get rid of this error here?

Comment (ismayilov-ismayil) [reply src/Agent.Listener/Telemetry/TelemetryPublisher.cs:31]: CA generates compile error due to disposable object
image

Comment (kirill-ivlev) [reply src/Agent.Listener/Telemetry/TelemetryPublisher.cs:31]: @ismayilov-ismayil I think you can use short `using`  syntax for vssConnection

Comment (ismayilov-ismayil) [reply src/Agent.Listener/Telemetry/TelemetryPublisher.cs:31]: Fixed

**Review by KonstantinTyukalov — APPROVED** [review]
[Suggestion] LGTM, but as I see, you copied the code from the Worker project. Is it possible to move all related stateless services to the SDK to reduce code duplication?

**Comment by ismayilov-ismayil** [comment]
> [Suggestion] LGTM, but as I see, you copied the code from the Worker project. Is it possible to move all related stateless services to the SDK to reduce code duplication?

Actually I did so, but since they have different credentials retrieving logic, only Interfaces can be merged into one place, though I separated them. I will try again, thanks for review and suggestion

**Review by LiliaSabitova — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Review by Pranshu-Negi — APPROVED** [review]

**Pull request merged**